Example/MDMCoreDataTests/MDMPersistenceControllerTests.m

```diff
@@ -182,4 +185,143 @@ - (void)testExecuteFetchRequest {
+ */
+- (void)testPrivateManagedObjectContextWithNewPersistentStoreCoordinator {
+    
+    NSManagedObjectContext *foregroundMOC = self.persistenceController.managedObjectContext;
+    //Register to receive background save notifications
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(backgroundManagedObjectContextDidSaveNotification:) name:MDMIndpendentManagedObjectContextDidSaveNotification object:nil];
+    
+    //Initialize expectation for receiving background save notification 
+    self.backgroundSaveNotificationExpectation = [self expectationWithDescription:@"Should have received background save notification"];
+    
+    //Create independent background context
+    NSManagedObjectContext * backgroundMOC = [self.persistenceController createPrivateManagedObjectContextWithNewPersistentStoreCoordinator];
+    //To test fail scenario - uncomment the below line to use a context that has the same PersistentStoreCoordinator as the foregroundMOC
+    //NSManagedObjectContext *backgroundMOC = [self.persistenceController performSelector:sel_getUid("writerObjectContext")];
+    XCTAssertNotNil(backgroundMOC, @"Should not fail creation of background context");
+    
+    //Create many objects on background context (wait for this to complete)
+    NSUInteger objCountBG =1000000;
+    [self createTestEntitiesAndWaitWithContext:backgroundMOC objectCount:objCountBG];
+    
+    //Start concurrent operations - one for background save and another for foreground fetch
+    //  Time taken for foreground retrieve should not exceed 1 second (observed around 0.0002 seconds
+    //  for fetch vs 12 seconds for saving 1 million Test entities on a Mac Book Pro 2.8GHz Intel Core
+    //  i7 based iPhone Simulator debug mode)
+    //  It is feasible to calibrate for the expected fetch time on current running device - but a max of 1 second should suffice for this test.
+    
+    //Concurrent Op 1: Background Save
+    [backgroundMOC performBlock:^{
+        NSLog(@"Background save start time %@", [NSDate date]);
+        NSError *backgroundMOCError;
+        [backgroundMOC save:&backgroundMOCError];
+        XCTAssertNil(backgroundMOCError, @"Should not fail saving background context changes");
+        NSLog(@"Background save completion time %@", [NSDate date]);
+    }];
+    
+    //Concurrent Op 2: Foreground Fetch
+    sleep(0.05); //let the background save begin
+    NSTimeInterval maxExpectedFetchTime = 1.0;
+    NSDate * multifetchStartTime = [NSDate date];
+    NSLog(@"Foreground Multiple Fetch start time %@", multifetchStartTime);
+    // Looping serves two purposes:
+    // 1) Accounts for cases where background save does not start immediately - though the sleep above should most likely suffice.
+    // 2) Demonstrating the difference in fetch times (before and after save completion) when using non-independent backgroundMOC.
```

Comment:
Ah gotcha! I guess my only 2 cents here would be: is there a way to make this more obvious or explicit? Might be worth making a whole separate test that expects the first fetch to occur after the background MOC finishes saving?